src/openzeppelin/token/erc1155/library.cairo

```diff
@@ -0,0 +1,601 @@
+        let (caller) = get_caller_address();
+        with_attr error_message("ERC1155: cannot call transfer from the zero address") {
+            assert_not_zero(caller);
+        }
+        assert_owner_or_approved(from_);
+        _safe_batch_transfer_from(from_, to, ids_len, ids, amounts_len, amounts, data_len, data);
+        return ();
+    }
+
+    //
+    // Internals
+    //
+
+    func _safe_transfer_from{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        from_: felt, to: felt, id: Uint256, amount: Uint256, data_len: felt, data: felt*
+    ) {
+        alloc_locals;
+        // Check args
+        with_attr error_message("ERC1155: transfer to the zero address") {
+            assert_not_zero(to);
+        }
+        _check_id(id);
+        with_attr error_message("ERC1155: amount is not a valid Uint256") {
+            uint256_check(amount);
+        }
+
+        // Deduct from sender
+        let (from_balance: Uint256) = ERC1155_balances.read(id, from_);
+        with_attr error_message("ERC1155: insufficient balance for transfer") {
+            let (new_balance: Uint256) = SafeUint256.sub_le(from_balance, amount);
+        }
+        ERC1155_balances.write(id, from_, new_balance);
+
+        // Add to receiver
+        _add_to_receiver(id, amount, to);
+
+        // Emit events and check
+        let (operator) = get_caller_address();
+        TransferSingle.emit(operator, from_, to, id, amount);
+
+        _do_safe_transfer_acceptance_check(operator, from_, to, id, amount, data_len, data);
+        return ();
+    }
+
+    func _safe_batch_transfer_from{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        from_: felt,
+        to: felt,
+        ids_len: felt,
+        ids: Uint256*,
+        amounts_len: felt,
+        amounts: Uint256*,
+        data_len: felt,
+        data: felt*,
+    ) {
+        alloc_locals;
+        // Check args
+        with_attr error_message("ERC1155: transfer to the zero address") {
+            assert_not_zero(to);
+        }
+        with_attr error_message("ERC1155: ids and amounts length mismatch") {
+            assert ids_len = amounts_len;
+        }
+        // Recursive call
+        _safe_batch_transfer_from_iter(from_, to, ids_len, ids, amounts);
+
+        // Emit events and check
+        let (operator) = get_caller_address();
+        TransferBatch.emit(operator, from_, to, ids_len, ids, amounts_len, amounts);
+
+        _do_safe_batch_transfer_acceptance_check(
+            operator, from_, to, ids_len, ids, amounts_len, amounts, data_len, data
+        );
+        return ();
+    }
+
+    func _mint{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        to: felt, id: Uint256, amount: Uint256, data_len: felt, data: felt*
+    ) {
+        // Cannot mint to zero address
+        with_attr error_message("ERC1155: mint to the zero address") {
+            assert_not_zero(to);
+        }
+        // Check uints validity
```

Comment:
```suggestion
```

For consistency, I would remove the comment from all the places or add it to all of them. For now, we have it in some places and not in others.

Reply:
you mean this specific uint comment or all comments throughout the file?

Reply:
This specific one.